xarray/core/duck_array_ops.py

```diff
@@ -181,16 +181,49 @@ def allclose_or_equiv(arr1, arr2, rtol=1e-5, atol=1e-8):
     arr2 = asarray(arr2)
     if arr1.shape != arr2.shape:
         return False
+    if (
+        dask_array
+        and isinstance(arr1, dask_array.Array)
+        and isinstance(arr2, dask_array.Array)
+    ):
+        # GH3068
+        if arr1.name == arr2.name:
+            return True
     return bool(isclose(arr1, arr2, rtol=rtol, atol=atol, equal_nan=True).all())
 
 
+def lazy_array_equiv(arr1, arr2):
+    """Like array_equal, but doesn't actually compare values
+    """
+    arr1 = asarray(arr1)
+    arr2 = asarray(arr2)
+    if arr1.shape != arr2.shape:
+        return False
+    if (
+        dask_array
+        and isinstance(arr1, dask_array.Array)
+        and isinstance(arr2, dask_array.Array)
+    ):
+        # GH3068
+        if arr1.name == arr2.name:
+            return True
+
+
 def array_equiv(arr1, arr2):
     """Like np.array_equal, but also allows values to be NaN in both arrays
     """
     arr1 = asarray(arr1)
     arr2 = asarray(arr2)
     if arr1.shape != arr2.shape:
         return False
+    if (
+        dask_array
+        and isinstance(arr1, dask_array.Array)
+        and isinstance(arr2, dask_array.Array)
+    ):
+        # GH3068
+        if arr1.name == arr2.name:
+            return True
```

Comment:
This is looking a little repetitive. Can you make a helper function for doing these checks?

Also note the similar logic (checking object identity) inside `Variable.equals`

Reply:
Done. We can just call `lazy_array_equiv` in all these before doing the actual value comparison. Then only the `asarray` cast is duplicated.

Re `Variable.equals`:  
https://github.com/pydata/xarray/blob/fb0cf7b5fe56519a933ffcecbce9e9327fe236a6/xarray/core/variable.py#L1560-L1576

`equiv=duck_array_ops.array_equiv` by default which now calls `duck_array_ops.lazy_array_equiv` first so things are good?

Reply:
OK I've added the identity check to `lazy_array_equiv` too